src/index.ts

```diff
@@ -125,11 +138,20 @@ export function wrap<
   makeCacheKey = (defaultMakeCacheKey as () => TCacheKey),
   keyArgs,
   subscribe,
-  cache = new StrongCache(
-    max,
-    entry => entry.dispose(),
-  ),
+  cache: cacheOption,
 }: OptimisticWrapOptions<TArgs, TKeyArgs, TCacheKey, TResult> = Object.create(null)) {
+  let cache: CommonCache<TCacheKey, Entry<TArgs, TResult>>;
+  if (cacheOption) {
+    cache = typeof cacheOption === "function"
+      ? new cacheOption(max)
+      : cacheOption;
+  } else {
+    cache = new StrongCache<TCacheKey, Entry<TArgs, TResult>>(
+      max,
+      entry => entry.dispose(),
+    );
+  }
```

Comment:
Since the `CommonCacheConstructor` type above requires the `dispose` argument anyways, why not call it in the `new cacheOption(max)` line, too? 
      
```suggestion
  let cache: CommonCache<TCacheKey, Entry<TArgs, TResult>>;
  if (cacheOption) {
    cache = typeof cacheOption === "function"
      ? new cacheOption(max, entry => entry.dispose(),)
      : cacheOption;
  } else {
    cache = new StrongCache<TCacheKey, Entry<TArgs, TResult>>(
      max,
      entry => entry.dispose(),
    );
  }
```

as a result, you could also simplify the whole thing if you changed the default value of `cacheOption` to  `StrongCache` up in the parameters.

```diff
  subscribe,
-  cache: cacheOption,
+  cache: cacheOption = StrongCache,
}: OptimisticWrapOptions<TArgs, TKeyArgs, TCacheKey, TResult> = Object.create(null)) {
```

```suggestion
  let cache: CommonCache<TCacheKey, Entry<TArgs, TResult>>;
  cache = typeof cacheOption === "function"
      ? new cacheOption(max, entry => entry.dispose(),)
      : cacheOption;
  }
```

Reply:
Great ideas! Done: f0133c0f95cb138ca71a96b35b4549695814a51e